src/main/java/games/strategy/engine/framework/map/download/DownloadFile.java

```diff
@@ -19,24 +16,21 @@
     NOT_STARTED, DOWNLOADING, CANCELLED, DONE
   }
 
-  private final List<Runnable> downloadCompletedListeners = new ArrayList<>();
-  private final Consumer<Long> progressUpdateListener;
-  private final DownloadFileDescription downloadDescription;
+  private final DownloadFileDescription download;
+  private final DownloadListener downloadListener;
   private volatile DownloadState state = DownloadState.NOT_STARTED;
 
-  /**
-   * Creates a new DownloadFile object.
-   * Does not actually start the download, call 'startAsyncDownload()' to start the download.
-   *
-   * @param download The details of what to download
-   * @param progressUpdateListener Called periodically while download progress is made.
-   * @param completionListener Called when the File download is complete.
-   */
-  DownloadFile(final DownloadFileDescription download, final Consumer<Long> progressUpdateListener,
-      final Runnable completionListener) {
-    this.downloadDescription = download;
-    this.progressUpdateListener = progressUpdateListener;
-    addDownloadCompletedListener(completionListener);
+  DownloadFile(final DownloadFileDescription download, final DownloadListener downloadListener) {
+    this.download = download;
+    this.downloadListener = downloadListener;
+
+    // TODO: consider running in a separate thread because this constructor is called while a lock is held
+    // it works now because the listener calls SwingUtilities.invokeLater(), but that may not always be the case
+    downloadListener.downloadStarted(download);
```

Comment:
Should we go ahead and address this TODO now?  Creating a new thread here to run a single callback is ugly, but it avoids the potential for deadlock in the future.  Otherwise, if we're content with the way the download code works now, I'd say remove the TODO.